### PR TITLE
Fix/flow to flow docs

### PR DESCRIPTION
**Pull request by AlessandroLollo**
## Summary

This PR is an attempt to fix issue https://github.com/PrefectHQ/prefect/issues/5092

Replaces StartFlowRun with create_flow_run in https://docs.prefect.io/core/idioms/flow-to-flow.html

## Importance
Keep docs up to date.
This PR:

- [ ] adds new tests (if appropriate)
- [ ] adds a change file in the `changes/` directory (if appropriate)
- [ ] updates docstrings for any new functions or function arguments, including `docs/outline.toml` for API reference docs (if appropriate)

### CODE REVIEW

File: docs/core/idioms/flow-to-flow.md [hunk 1]
```diff
@@ -85,10 +85,10 @@ weekday_schedule = CronSchedule(
...
 
 # assumes you have registered the following flows in a project named "examples"
-flow_a = StartFlowRun(flow_name="A", project_name="examples", wait=True)
-flow_b = StartFlowRun(flow_name="B", project_name="examples", wait=True)
-flow_c = StartFlowRun(flow_name="C", project_name="examples", wait=True)
-flow_d = StartFlowRun(flow_name="D", project_name="examples", wait=True)
+flow_a = create_flow_run(flow_name="A", project_name="examples", wait=True)
+flow_b = create_flow_run(flow_name="B", project_name="examples", wait=True)
+flow_c = create_flow_run(flow_name="C", project_name="examples", wait=True)
+flow_d = create_flow_run(flow_name="D", project_name="examples", wait=True)
```
Comment (madkinsz) [inline docs/core/idioms/flow-to-flow.md:91]: There's a separate `wait_for_flow_run` task, there's no `wait` setting for `create_flow_run`

Comment (AlessandroLollo) [reply docs/core/idioms/flow-to-flow.md:91]: Ah right!
In this example, should we just show how to use `create_flow_run`, or should we also explain how `wait_for_flow_run`?

Comment (madkinsz) [reply docs/core/idioms/flow-to-flow.md:91]: I think if updating the examples we should probably include `wait_for_flow_run`

File: docs/core/idioms/flow-to-flow.md [hunk 2]
```diff
@@ -63,9 +64,9 @@ with Flow("parent-flow") as flow:
 ## Scheduling a Flow-of-Flows
 
 Oftentimes different people are responsible for maintaining different flows; in this case it can be useful
-to construct a Flow-of-Flows that specifies stateful dependencies between various Flows.  The `wait` kwarg
-allows you to specify that the task should wait until the triggered flow run completes, and reflect the
-flow run state as the task state.
+to construct a Flow-of-Flows that specifies stateful dependencies between various Flows.  
+The `wait_for_flow_run` task allows you to specify that the task should wait until the triggered flow run completes, 
+and reflect the flow run state as the task state.
```
Comment (madkinsz) [inline docs/core/idioms/flow-to-flow.md:69]:
```suggestion
The `wait_for_flow_run` task allows you to specify that the task should wait until the triggered flow run completes.
```

This task does not raise the state of the child flow like `StartFlowRun(wait=True)` did.

Comment (madkinsz) [reply docs/core/idioms/flow-to-flow.md:69]: Does the rest of the example make sense with this caveat?

Comment (AlessandroLollo) [reply docs/core/idioms/flow-to-flow.md:69]: Oh this is interesting!
So, I would I replicate the `StartFlowRun` behaviour using `creat_flow_run` and `wait_for_flow_run` ?

Comment (madkinsz) [reply docs/core/idioms/flow-to-flow.md:69]: It'll wait for the final state and it'll return a `FlowRunView` containing the final state, but you'd need a third task that raises the status (or we could add a bool to `wait_for_flow_run` which may make sense).

Comment (AlessandroLollo) [reply docs/core/idioms/flow-to-flow.md:69]: We'll follow your suggestion regarding `FlowRunView`.
However, I think that having a bool in `wait_for_flow_run` would greatly improve its usability.

Comment (AlessandroLollo) [reply docs/core/idioms/flow-to-flow.md:69]: > Does the rest of the example make sense with this caveat?

Probably not, I'm going to remove it.
I'll also accept the suggested change above.

Comment (AlessandroLollo) [reply docs/core/idioms/flow-to-flow.md]: I ended up accepting the suggested change, with a small adjustment that imho makes the rest of example useful.

**Comment by madkinsz** [comment]
Previewable at https://deploy-preview-5095--prefect-docs.netlify.app/core/idioms/flow-to-flow.html

@AlessandroLollo a bunch of these tasks are being called outside of the `with Flow` block and will fail.

**Comment by AlessandroLollo** [comment]
> Previewable at https://deploy-preview-5095--prefect-docs.netlify.app/core/idioms/flow-to-flow.html
> 
> @AlessandroLollo a bunch of these tasks are being called outside of the `with Flow` block and will fail.

Looking at the preview docs I see all of the inside the `with Flow` block...
What tasks do you see outside the flow block?

File: docs/core/idioms/flow-to-flow.md [hunk 3]
```diff
@@ -85,14 +86,21 @@ weekday_schedule = CronSchedule(
...
+flow_b = create_flow_run(flow_name="B", project_name="examples", wait=True)
+wait_for_flow_b = wait_for_flow_run(flow_b, raise_final_state=True)
+
+flow_c = create_flow_run(flow_name="C", project_name="examples", wait=True)
+wait_for_flow_c = wait_for_flow_run(flow_c, raise_final_state=True)
+
+flow_d = create_flow_run(flow_name="D", project_name="examples", wait=True)
+wait_for_flow_d = wait_for_flow_run(flow_d, raise_final_state=True)
 
 with Flow("parent-flow", schedule=weekday_schedule) as flow:
```
Comment (madkinsz) [inline docs/core/idioms/flow-to-flow.md:101]: These are all outside the flow block and have `wait=True` on `create_flow_run`

File: docs/core/idioms/flow-to-flow.md [hunk 4]
```diff
@@ -110,16 +118,23 @@ weekday_schedule = CronSchedule(
...
```
Comment (madkinsz) [inline docs/core/idioms/flow-to-flow.md:133]: These will fail as well.

**Review by madkinsz — APPROVED** [review]
Thanks @AlessandroLollo !

**Pull request merged**